streamer/src/quic.rs

```diff
@@ -0,0 +1,188 @@
+use {
+    crossbeam_channel::Sender,
+    futures_util::stream::StreamExt,
+    pem::Pem,
+    pkcs8::{der::Document, AlgorithmIdentifier, ObjectIdentifier},
+    quinn::{Endpoint, EndpointConfig, ServerConfig},
+    rcgen::{CertificateParams, DistinguishedName, DnType, SanType},
+    solana_perf::packet::PacketBatch,
+    solana_sdk::{
+        packet::{Packet, PACKET_DATA_SIZE},
+        signature::Keypair,
+    },
+    std::{
+        error::Error,
+        net::{IpAddr, UdpSocket},
+        sync::{
+            atomic::{AtomicBool, Ordering},
+            Arc,
+        },
+        thread,
+        time::Duration,
+    },
+    tokio::{
+        runtime::{Builder, Runtime},
+        time::timeout,
+    },
+};
+
+/// Returns default server configuration along with its PEM certificate chain.
+#[allow(clippy::field_reassign_with_default)] // https://github.com/rust-lang/rust-clippy/issues/6527
+fn configure_server(
+    identity_keypair: &Keypair,
+    gossip_host: IpAddr,
+) -> Result<(ServerConfig, String), Box<dyn Error>> {
+    let (cert_chain, priv_key) = new_cert(identity_keypair, gossip_host)?;
+    let cert_chain_pem_parts: Vec<Pem> = cert_chain
+        .clone()
+        .into_iter()
```

Comment:
Seems clearer to do a `.iter()`, and then move the clone into `cert.0.clone()` inside